### PR TITLE
More basic information about each service.

### PR DESCRIPTION
**Pull request by jlevy**
Add standard AWS links in each section, including all FAQs.
A bit more basic information about several services.
A new common section, and a link to the General Reference.
A few other small fixups.

### CODE REVIEW

**Comment by jlevy** [comment]
Fixes #28

**Comment by jlevy** [comment]
@ThanosBaskous review?

**Comment by jlevy** [comment]
Fixes #30, just a placeholder; still a long way to go on this topic!

File: README.md [hunk 1]
```diff
@@ -177,6 +179,14 @@ General Information
...
 -	[AWS in Plain English](https://www.expeditedssl.com/aws-in-plain-english) offers more friendly explanation of what all the other different services are.
 
+### Common Concepts
+
+-	📒 The AWS [**General Reference**](https://docs.aws.amazon.com/general/latest/gr/Welcome.html) covers a bunch of common concepts that are relevant for multiple services.
+-	AWS allows deployments in [**regions**](https://docs.aws.amazon.com/general/latest/gr/rande.html), which are isolated geographic locations that help you reduce latency or offer additional redundancy (though typically availability zones are the first tool of choice for [high availability](#high-availability)).
+-	Each service has **endpoints** that are typically different for each service.
+-	Endpoints differ from service to service and not all services are available in each region, as listed in [these tables](https://docs.aws.amazon.com/general/latest/gr/rande.html).
+-	[**Amazon Resource Names (ARNs)**](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) are specially formatted identifiers for identifying resources. They start with 'arn:' and used in many services, and in particular for IAM policies.
```
Comment (ThanosBaskous) [inline README.md]: Missing word? "and are used"

File: README.md [hunk 2]
```diff
@@ -454,15 +468,28 @@ We cover security basics first, since configuring user accounts is something you
...
 -	❗**Don’t share user credentials.** It’s remarkably common for first-time AWS users create one account and one set of credentials (access key or password), and then use them for a while, sharing among engineers and others within a company. This is easy. But *don’t do this*. This is an insecure practice for many reasons, but in particular, if you do, you will have reduced ability to revoke credentials on a per-user or per-service basis (for example, if an employee leaves or a key is compromised), which can lead to serious complications.
+-	🔸Some IAM operations are slower than other API calls (many seconds), since AWS needs to propagate these globally across regions.
 
 S3
 --
 
+### Basics
+
+-	📒 [Homepage](https://aws.amazon.com/s3/) ∙ [Developer guide](https://docs.aws.amazon.com/AmazonS3/latest/dev/Welcome.html) ∙ [FAQ](https://aws.amazon.com/s3/faqs/)
+-	**S3** (Simple Storage Service) is AWS’ standard cloud storage service, offering file (blob) storage of arbitrary number of files of small or large size (from 0 to 5 terabytes).
```
Comment (ThanosBaskous) [inline README.md]: nit: BLOB

File: README.md [hunk 3]
```diff
@@ -454,15 +468,28 @@ We cover security basics first, since configuring user accounts is something you
...
+-	Items, or **objects**, are placed into named **buckets** stored with names which are usually called **keys**. The main content is the **value**.
+-	Objects are created, deleted, or updated. Large objects can be streamed, but you cannot access or modify parts of a value; you need to update the whole object.
+-	Every object also has [**metadata**](https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html), which includes arbitrary key-value pairs, and is used in a way similar to HTTP headers. Some metadata is system-defined, some are significant when serving HTTP content from buckets or CloudFront, and you can also define arbitrary metadata for your own use.
+-	Although often bucket and key names are provided in APIs individually, it’s also common practice to write an S3 location in the form 's3://bucket-name/path/to/key' (where the key here is 'path/to/key'). (You’ll also see 's3n://' and 's3a://' prefixes [in Hadoop systems](https://wiki.apache.org/hadoop/AmazonS3).)
```
Comment (ThanosBaskous) [inline README.md]: I like these notes on alternatives and tradeoffs, especially for a heavily used general purpose service like S3. Maybe a pattern to encourage.

File: README.md [hunk 4]
```diff
@@ -454,15 +468,28 @@ We cover security basics first, since configuring user accounts is something you
...
+
 ### Tips
 
 -	For most practical purposes, you can consider S3 capacity unlimited, both in total size of files and number of objects.
--	S3 buckets use a **global naming scheme**, so if another AWS has already created a bucket under a name that you want to use you will need to pick a different name. A common practice is to use the company name acronym or abbreviation to prefix all bucket names (but please, don’t use this as a security measure).
+-	**Bucket naming:** Buckets are chosen from a global namespace (across all regions, even though S3 itself stores data in [whichever S3 region](https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region) you select), so you’ll find many bucket names are already taken. Creating a bucket means taking ownership of the name until you delete it. Bucket names have [a few restrictions](https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html) on them.
+	-	A common practice is to use the company name acronym or abbreviation to prefix all bucket names (but please, don’t be stupid and use a check on a bucket prefix as a security measure).
```
Comment (ThanosBaskous) [inline README.md]: A good note - is there a way to say it without "stupid"?

Taking a crack at it:
"A common practice is to use the company name acronym or abbreviation to prefix all bucket names (but keep in mind that checking the bucket prefix as a security measure is insufficient and easily circumvented)."

File: README.md [hunk 5]
```diff
@@ -675,6 +715,12 @@ ELBs
 Elastic IPs
 -----------
 
+### Basics
+
+-	📒 [Documentation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html) ∙ [FAQ](https://aws.amazon.com/ec2/faqs/#Elastic_IP)
+-	**Elastic IPs** are static IP addresses you can purchase from AWS to assign to EC2 instances.
```
Comment (ThanosBaskous) [inline README.md]: s/purchase/rent/

File: README.md [hunk 6]
```diff
@@ -737,7 +794,8 @@ ECS
 
 ### Basics
 
--	[ECS](https://aws.amazon.com/ecs/) (EC2 Container Service) is a relatively new service (launched end of 2014) that manages clusters of services deployed via Docker.
+-	📒 [Homepage](https://aws.amazon.com/ecs/) ∙ [Developer guide](http://docs.aws.amazon.com/AmazonECS/latest/developerguide/) ∙ [FAQ](https://aws.amazon.com/ecs/faqs/)
+-	**ECS** (EC2 Containser Service) is a relatively new service (launched end of 2014) that manages clusters of services deployed via Docker.
```
Comment (ThanosBaskous) [inline README.md]: Typo: "Containser".

File: README.md [hunk 7]
```diff
@@ -901,20 +1023,23 @@ EMR
...
 	-	Deploy instances evenly across all available AZs, so that only a minimal fraction of your capacity is lost in case of an AZ outage.
 	-	If your architecture has single points of failure, put all of them into a single AZ. This may seem counter-intuitive, but it minimizes the likelihood of any one SPOF to go down on an outage of a single AZ.
 -	**EBS vs instance storage:** For a number of years, EBSs had a poorer track record for availability than instance storage. For systems where individual instances can be killed and restarted easily, instance storage with sufficient redundancy could give higher availability overall. EBS has improved, and modern instance types (since 2015) are now EBS-only, so this approach, while helpful at one time, may be increasingly archaic.
--	Be sure you use and understand **ELBs** whenever appropriate. (See the section on ELBs.) Many outages are due to not using load balancers, or misunderstandings or misconfigurations of ELBs.
+-	Be sure to [use and understand ELBs](#elbs) appropriately. Many outages are due to not using load balancers, or misunderstandings or misconfigurations of ELBs.
```
Comment (ThanosBaskous) [inline README.md]: Any citations, examples, or gotchas?

Aggressive healthchecks? Autoscaler termination policies tied to ELB healthcheck status? Pointing DNS to A record of current IP of an ELB shard that ends up changing?

**Comment by ThanosBaskous** [comment]
I had a few notes, but very good stuff overall, @jlevy!

File: README.md (hunk 2 above)
Comment (jlevy) [inline README.md]: Huh, you got me wondering on what's correct nowadays. :) I've always preferred “blob”. https://en.wikipedia.org/wiki/Binary_large_object

File: README.md (hunk 7 above)
Comment (jlevy) [inline README.md]: The last one is already there, but I'm emphasizing it a bit more. You want to take a crack at adding the other two to the ELB section — those are good ones. #33

File: README.md (hunk 3 above)
Comment (jlevy) [inline README.md]: Yeah, agreed. Other ideas of ones to cover?

**Comment by jlevy** [comment]
👍 Thanks for the review!
